test/gtest/ucp/test_ucp_tag.cc

```diff
@@ -503,26 +503,48 @@ class test_ucp_tag_limits : public test_ucp_tag {
         check_short_thresh(cfg->am_u.max_reply_eager_short, cfg_thresh);
     }
 
-    void check_ep_proto_rndv_v2(size_t cfg_thresh, bool expect_rndv)
+
+    size_t get_supported_msg_size(ucp_proto_select_param_t select_param,
+                                  size_t cfg_thresh)
     {
-        ucp_ep_config_t *cfg             = ucp_ep_config(sender().ep());
-        ucp_proto_select_t *proto_select = &cfg->proto_select;
-        size_t msg_length                = cfg_thresh;
-        ucp_proto_select_elem_t value;
+        uint64_t tag_op_id_mask = UCS_BIT(UCP_OP_ID_TAG_SEND) |
+                                  UCS_BIT(UCP_OP_ID_TAG_SEND_SYNC);
 
-        if (m_test_offload) {
+        if (m_test_offload &&
+            ucp_proto_select_check_op(&select_param, tag_op_id_mask)) {
             /* There is a lower bound for rndv threshold with tag offload.
              * We should not send messages smaller than this limit */
-            msg_length = ucs_max(m_min_rndv, msg_length);
+            return std::max(m_tag_min_rndv, cfg_thresh);
         }
+        return cfg_thresh;
+    }
+
+    void check_ep_proto_rndv_v2(size_t cfg_thresh, bool expect_rndv)
+    {
+        ucp_ep_config_t *cfg             = ucp_ep_config(sender().ep());
+        ucp_proto_select_t *proto_select = &cfg->proto_select;
+        const ucp_proto_config_t *proto_config;
+        ucp_proto_select_elem_t value;
+        size_t msg_length;
 
         kh_foreach_value(proto_select->hash, value, {
             /* Find index of the corresponding ucp_proto_threshold_elem_t
              * to handle the given message size */
-            unsigned idx = 0;
-            const ucp_proto_config_t *proto_config;
+            unsigned idx  = 0;
+            proto_config  = &value.thresholds[idx].proto_config;
+            msg_length    = get_supported_msg_size(proto_config->select_param,
+                                                   cfg_thresh);
+
             for (; msg_length > value.thresholds[idx].max_msg_length; ++idx) {
                 proto_config = &value.thresholds[idx].proto_config;
+                /* Skip check that rndv is disabled for tag offload use case.
+                 * With tag offload rndv protocol might still be used for large
+                 * messages, because of HWTM limit for eager transfers.
+                 * e.g. RNDV_THRESH=inf + HWTM */
+                if((m_test_offload) &&
+                   (value.thresholds[idx].max_msg_length >= m_tag_min_rndv)) {
```

Comment:
This check is true but too relaxed, here should be something like:
```c
    (value.thresholds[idx].max_msg_length >= sender().iface_attr().cap.tag.eager.max_zcopy)) {
```

But there is no simple way to get this `max_zcopy` value on UCP layer. So it was decided to completely disable this test in case when HWTM is enabled.